server/sms/nexmo_sms_sender.go

```diff
@@ -150,39 +202,52 @@ func (ns *NexmoSender) Send(msg *protocol.Message) error {
 			Jitter: true,
 		},
 	}
-
-	err = withRetry.executeAndCheck(sendSms)
-	if err != nil && err == ErrRetryFailed {
```

Comment:
I guess it's fine that I removed the first condition err != nil, it was redundant